plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -0,0 +1,399 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.plugin.jdbc.BaseJdbcConnectorTest;
+import io.trino.spi.connector.SortOrder;
+import io.trino.sql.planner.assertions.PlanMatchPattern;
+import io.trino.sql.planner.plan.TableScanNode;
+import io.trino.sql.planner.plan.TopNNode;
+import io.trino.sql.query.QueryAssertions;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorBehavior;
+import io.trino.testing.sql.SqlExecutor;
+import io.trino.testing.sql.TestTable;
+import org.intellij.lang.annotations.Language;
+
+import java.util.List;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.plugin.ignite.IgniteQueryRunner.createIgniteQueryRunner;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.node;
+import static io.trino.testing.TestingConnectorBehavior.SUPPORTS_TOPN_PUSHDOWN_WITH_VARCHAR;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestIgniteConnectorTest
+        extends BaseJdbcConnectorTest
+{
+    private TestingIgniteServer igniteServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.igniteServer = closeAfterClass(new TestingIgniteServer());
+        return createIgniteQueryRunner(
+                igniteServer,
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                REQUIRED_TPCH_TABLES);
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_DELETE:
+                return false;
+
+            case SUPPORTS_CREATE_SCHEMA:
+            case SUPPORTS_CREATE_TABLE:
+            case SUPPORTS_CREATE_TABLE_WITH_DATA:
+            case SUPPORTS_CREATE_VIEW:
+            case SUPPORTS_RENAME_TABLE:
+            case SUPPORTS_COMMENT_ON_TABLE:
+            case SUPPORTS_COMMENT_ON_COLUMN:
+            case SUPPORTS_ARRAY:
+                return false;
+
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    public void testInsert()
```

Comment:
Instead of copying tests, can we somehow change tests so they could allow to add custom table properties?

Reply:
That would need to change the method ``AbstractTestDistributedQueries#testInsert``.
I think maybe we can create another pr for this purpose.

Reply:
Another thing is that, no only added table properties, but also avoid insert into duplicate `primary key` for the table in this tests. 
If we want to change base tests, we need to discuss about that too.

Reply:
I am afraid that every time one add a new test they will fail an issue in Ignite connector as it requires dedicated care. So we will force people to deal with Ignite connector if if they don't have time or competencies (it requires time to learn that Ignite connector has table properties and it does not allow `primary key` duplicates).

I don't know how to deal with `primary key` duplicates issues, possibly it requires us to override.  But properties handling is mechanical and could be done automatically. Changing `AbstractTestDistributedQueries` is fine as long it is backward compatible change.